### ISSUE TITLE
Lack of support for the refactoring using conditional expression transformation

### ISSUE THREAD

**Issue by simpt5** [tsantalis/RefactoringMiner#681]
## Tested tools and version:
Refactoring Aware Commit Review Chrome Extension (Version: 2.0.4).
RefactoringMiner (Version 3.0.4) maven dependency.

## Description:
During the usage of RefactoringMiner, I found a refactoring case using conditional expression transformation that can’t be detected. Could it possibly be supported in the future maybe?

Commit URL: https://github.com/dromara/hutool/pull/1109/commits/1d4ac7d0dca04ff8b6d64b4c47cc5ff01d60201d

File URL: https://github.com/dromara/hutool/pull/1109/commits/1d4ac7d0dca04ff8b6d64b4c47cc5ff01d60201d#diff-e9a28972da0c69b8a3c84ac4d14c9c911bb58f552aef78049f00909468cbf1be

The following refactoring for conditional expression in the above commit can’t be detected.
 
![image](https://github.com/tsantalis/RefactoringMiner/assets/163680970/6f782bb0-d568-414c-adb8-8b492b002823)

**Comment by tsantalis** [comment]
@simpt5 
The corresponding if statements are actually matched. I am not sure if this is a refactoring type.
It can be considered as [Simplifying Boolean Expressions](http://cs111.wellesley.edu/~cs111/archive/cs111_spring00/public_html/lectures/boolean-simplification.html)

![image](https://github.com/tsantalis/RefactoringMiner/assets/1483516/09c9f78e-4021-4264-b524-de09547116ec)

**Comment by simpt5** [comment]
Yes, it is actually can be treated as a boolean expression transformation, which transform boolean expression using some rules without changing the program behaviors. Since those transformations are based on rules and performed on the boolean expression, I am wondering if those could be detected as refactoring maybe?

**Comment by tsantalis** [comment]
@simpt5 
It is definitely possible to implement this in RefactoringMiner and get it reported with a new refactoring type `Simplify Boolean Expression`.
I can add it as a new feature request.
